pandas/core/groupby.py

```diff
@@ -3964,13 +4012,117 @@ def filter(self, func, dropna=True, *args, **kwargs):  # noqa
 
 class DataFrameGroupBy(NDFrameGroupBy):
     _apply_whitelist = _dataframe_apply_whitelist
```

Comment:
i would remove the whitelist entirely; since you are defining the whitelist the methods no longer are referenced thru that mechanism (and instead thru explicit methods).  you can simply setup a list in the tests that is a list of these methods and check that they are defined.